appshell/cefclient_gtk.cpp

```diff
@@ -18,47 +20,101 @@
 #include "client_switches.h"
 #include "appshell_node_process.h"
 
-static std::string APPICONS[] = {"appshell32.png","appshell48.png","appshell128.png","appshell256.png"};
-char szWorkingDir[512];  // The current working directory
+static std::string APPICONS[] = {"appshell32.png", "appshell48.png", "appshell128.png", "appshell256.png"};
+char szWorkingDir[512]; // The current working directory
 std::string szInitialUrl;
 std::string szRunningDir;
 int add_handler_id;
 bool isReallyClosing = false;
 
+static gint DEFAULT_SIZE_HEIGHT = 600;
+static gint DEFAULT_SIZE_WIDTH = 800;
+
 // The global ClientHandler reference.
 extern CefRefPtr<ClientHandler> g_handler;
 
 // Application startup time
 time_t g_appStartupTime;
 
+// Error handler for X11
+
+int XErrorHandlerImpl(Display *display, XErrorEvent *event) {
+  LOG(WARNING)
+          << "X error received: "
+          << "type " << event->type << ", "
+          << "serial " << event->serial << ", "
+          << "error_code " << static_cast<int> (event->error_code) << ", "
+          << "request_code " << static_cast<int> (event->request_code) << ", "
+          << "minor_code " << static_cast<int> (event->minor_code);
+  return 0;
+}
+
+int XIOErrorHandlerImpl(Display *display) {
+  return 0;
+}
+
 void destroy(void) {
-  CefQuitMessageLoop();
+  g_message("destroy called");
+
+  if (isReallyClosing) {
+    CefQuitMessageLoop();
+  }
 }
 
-void TerminationSignalHandler(int signatl) {
+void TerminationSignalHandler(int signal) {
+  isReallyClosing = true;
   destroy();
 }
-void HandleAdd(GtkContainer *container,
-               GtkWidget *widget,
-               gpointer user_data) {
-  g_signal_handler_disconnect(container, add_handler_id);
-  if(gtk_widget_get_can_focus(widget)) {
-    gtk_widget_grab_focus(widget);
+
+//void HandleAdd(GtkContainer *container,
+//        GtkWidget *widget,
+//        gpointer user_data) {
+//  g_signal_handler_disconnect(container, add_handler_id);
+//  if (gtk_widget_get_can_focus(widget)) {
+//    gtk_widget_grab_focus(widget);
+//  } else {
+//    add_handler_id = g_signal_connect(G_OBJECT(widget), "add",
+//            G_CALLBACK(HandleAdd), NULL);
+//  }
+//}
+
+gboolean HandleQuit2(GtkWidget* widget, GdkEvent* event, GtkWindow* window) {
+  g_message("HandleQuit called");
```

Comment:
@ingorichter @jasonsanjose I think we should get rid off all the `g_message()` calls in this file.